Java-Introduction.md

```diff
@@ -0,0 +1,123 @@
+
+Platform specific installation instructions
+
+### Windows
+
+* Download the relevant .msi file (x86 / i586 for 32bits, x64 for 64bits)
+* Run the .msi file. Its a self extracting executable file which will install Java in your system!
+
+### Linux
+
+* Download the relevant tar.gz file for your system and install :
+
+   ```bash
+   $ tar zxvf jdk-8uversion-linux-x64.tar.gz
+   ```
+* RPM based Linux platforms download the relevant .rpm file and install :
+
+   ```bash
+   $ rpm -ivh jdk-8uversion-linux-x64.rpm
+   ```
+* Users have the choice to install an open source version of Java, OpenJDK or the Oracle JDK. While OpenJDK is in active development and in sync with Oracle JDK, they just differ in licensing stuff, however few developers complain of the stability of Open JDK. Instructions for **Ubuntu** :
+
+   Open JDK installation :
+   ```bash
+   sudo apt-get install openjdk-8-jdk
+   ```
+
+   Oracle JDK installation :
+   ```bash
+   sudo add-apt-repository ppa:webupd8team/java
+   sudo apt-get update
+   sudo apt-get install oracle-java8-installer
+   ```
+
+### Mac
+
+* Either download Mac OSX .dmg executable from Oracle Downloads
+* Or use [Homebrew](http://brew.sh/) to [install](http://stackoverflow.com/a/28635465/2861269) :
+
+   ```bash
+   brew tap caskroom/cask
+   brew install brew-cask
+   brew cask install java
+   ```
+
+### Verify Installation
+
+Verify Java has been properly installed in your system by opening Command Prompt / Terminal and checking the versions of Java runtime and compiler :
+
+```bash
+$ java -version
+java version "1.8.0_66"
+Java(TM) SE Runtime Environment (build 1.8.0_66-b17)
+Java HotSpot(TM) 64-Bit Server VM (build 25.66-b17, mixed mode)
+
+$ javac -version
+javac 1.8.0_66
+```
+
+## JVM
+
+Ok now since we are done with the installations, lets begin to understand first the nitty gritty of the Java ecosystem. Java is an interpreted language, that is the code we write gets compiled and then run. We write the code in .java files, Java compiles them into [bytecodes](https://en.wikipedia.org/wiki/Java_bytecode) which are run on a Java Virtual Machine or JVM for execution. These bytecodes typically has a .class extension.
+
+Java is a pretty secured language as it doesn't let your program run directly on the machine. Instead, your program runs on a Virtual Machine called JVM. This Virtual Machine exposes several API's for low level machine interactions you can make, but other than that you cannot play with machine instructions explicitely. This adds a huge bonus of security.
+
+Also, once your bytecode is compiled it can run on any Java VM. This Virtual Machine is machine dependent, i.e it has different implementations for Windows, Linux and Mac. But your program is guranteed to run in any system thanks to this VM. This philosophy is called ["Write Once, Run Anywhere"](https://en.wikipedia.org/wiki/Write_once,_run_anywhere).
+
```

Comment:
I always imagined having a diagram to explain this.

Remind me to explain this - but for now, ignore this line comment. You don't need to add it in this commit.